### DISCUSSION TITLE
SSL for Preview Adonis Version

### DISCUSSION THREAD

**Discussion by quangthienit1994** [adonisjs/core#1459] [Help] [answered]
I need add SSL to my project running on Adonis 5. How can  I do it ?

**Comment by popovicieduard** [comment, marked as answer]
SSL has nothing to do with adonisjs.

Issuing a ssl with: https://www.sslforfree.com/ or with certbot: https://certbot.eff.org/
And then in your nginx config block or apache set the cert path.

Here is a tutorial that can give you a clearer answer: https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-14-04

Or check the adonis js section: https://adonisjs.com/recipes/4.1/using-https. (this is for 4.1 but something similar can be done I think, NOT SURE)

Aaand another options is to use cloudflare DNS.

**Reply by MitchellMcKenna** [reply]
Adapting the self-signed recipe for local https development from 4.1 to 5.0, I think this is working:
```js
import 'reflect-metadata'
import sourceMapSupport from 'source-map-support'
import { Ignitor } from '@adonisjs/core/build/src/Ignitor'
import https from 'https'
import pem from 'pem'

sourceMapSupport.install({ handleUncaughtExceptions: false })

pem.createCertificate({ days: 1, selfSigned: true }, (error, keys) => {
  if (error) {
    return console.log(error)
  }

  const options = {
    key: keys.serviceKey,
    cert: keys.certificate
  }

  new Ignitor(__dirname)
    .httpServer()
    .start((handler) => {
      return https.createServer(options, handler)
    })
    .catch(console.error)
})
```
I had to enable this flag: chrome://flags/#allow-insecure-localhost to allow access to localhost. And I had to type "thisisunsafe" in the tab as mentioned here: https://support.google.com/chrome/thread/10551759?hl=en

**Reply by quangthienit1994** [reply]
Thank your answer, but I used forward Port of Apache or Ngnix.
